src/components/document/document-content.tsx

```diff
@@ -53,25 +54,42 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   private domElement: HTMLElement | null;
   private rowRefs: Array<TileRowComponent | null>;
   private mutationObserver: MutationObserver;
+  private scrollDisposer: IReactionDisposer;
 
   public componentDidMount() {
     if (this.domElement) {
       this.domElement.addEventListener("scroll", throttle(this.updateVisibleRows, 100));
       this.updateVisibleRows();
       this.scrollToSection(this.props.selectedSectionId);
 
-      if ((window as any).MutationObserver) {
-        this.mutationObserver = new MutationObserver(this.handleRowElementsChanged);
-        this.mutationObserver.observe(this.domElement, { childList: true });
-      }
       // We pass the domElement to our children, but it's undefined during the first render,
       // so we force an update to make sure we draw at least once after we have our domElement.
       this.forceUpdate();
+
+      this.scrollDisposer = reaction(
+        () => ({ tileId: this.stores.ui.scrollTo?.tileId, docId: this.stores.ui.scrollTo?.docId }),
+        (scrollTo: Record<string, any>, prevScrollTo: Record<string, any>) => {
+        if (scrollTo.tileId && this.props.content?.documentIdentifier === scrollTo.docId) {
+          this.rowRefs.forEach((row: TileRowComponent | null) => {
+            if (row?.tileRowDiv && row.hasTile(scrollTo.tileId)) {
+              // Javascript struggles to scroll multiple elements at the same time,
```

Comment:
More precise would be to say something like `Browsers will only animate one scroll at a time,`.